packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceProjectCreate.test.ts

```diff
@@ -86,41 +87,51 @@ describe('Force Project Create', () => {
   });
 
   describe('SelectProjectName Gatherer', () => {
-    let inputBoxSpy: sinon.SinonStub;
+    let gatherer: SelectProjectName;
+    let inputBox: sinon.SinonStub;
 
-    before(() => {
-      inputBoxSpy = sinon.stub(vscode.window, 'showInputBox');
-      inputBoxSpy.onCall(0).returns(undefined);
-      inputBoxSpy.onCall(1).returns('');
-      inputBoxSpy.onCall(2).returns(PROJECT_NAME);
+    beforeEach(() => {
+      gatherer = new SelectProjectName();
+      inputBox = sinon.stub(vscode.window, 'showInputBox');
     });
 
-    after(() => {
-      inputBoxSpy.restore();
+    afterEach(() => {
+      inputBox.restore();
+    });
+
+    it('Should make one call to showInputBox', async () => {
+      inputBox.returns(undefined);
+      const response = await gatherer.gather();
+      expect(inputBox.calledOnce).to.be.true;
     });
 
     it('Should return cancel if project name is undefined', async () => {
-      const gatherer = new SelectProjectName();
+      inputBox.returns(undefined);
       const response = await gatherer.gather();
-      expect(inputBoxSpy.calledOnce).to.be.true;
       expect(response.type).to.equal('CANCEL');
     });
 
     it('Should return cancel if user input is empty string', async () => {
-      const gatherer = new SelectProjectName();
+      inputBox.returns('');
       const response = await gatherer.gather();
-      expect(inputBoxSpy.calledTwice).to.be.true;
       expect(response.type).to.equal('CANCEL');
     });
 
     it('Should return Continue with inputted project name if project name is not undefined or empty', async () => {
-      const gatherer = new SelectProjectName();
+      inputBox.returns(PROJECT_NAME);
       const response = await gatherer.gather();
-      expect(inputBoxSpy.calledThrice).to.be.true;
+      expect(response.type).to.equal('CONTINUE');
+      if (response.type === 'CONTINUE') {
```

Comment:
this if isn't necessary with the above expect.  The test will fail and bail if type doesn't equal Continue.

Reply:
same on line 120

Reply:
Removing the check for response type gives a compile-time error.  Do you know how I can get around it? 
<img width="1404" alt="2022-03-25_10-46-09" src="https://user-images.githubusercontent.com/46458081/160174373-d1876248-0c98-4392-a25a-bcdf9b7ae77f.png">

Reply:
Resolved with Jeff's help (thanks @jeffb-sfdc!).  Was able to remove the conditional.